packages/reactive-element/src/reactive-element.ts

```diff
@@ -1140,7 +1153,13 @@ export abstract class ReactiveElement
         const shadowedProperties: string[] = [];
         (this.constructor as typeof ReactiveElement).elementProperties.forEach(
           (_v, p) => {
-            if (this.hasOwnProperty(p) && !this.__instanceProperties?.has(p)) {
+            if (
```

Comment:
Rather than loop over `elementProperties`, this can just loop over `prototypeToReactivePropertyNames.get(this.constructor.prototype)`, right?

Reply:
Yeah, I think you're right: in `createProperty`, the `elementProperties` entry is always set before the `prototypeToReactivePropertyNames` entry might be set, so `prototypeToReactivePropertyNames` should always be the subset we want here. Updated.